code/lib/router/src/router.tsx

```diff
@@ -48,7 +48,12 @@ export const useNavigate = () => {
 
   return useCallback((to: R.To | number, { plain, ...options } = {} as NavigateOptions) => {
     if (typeof to === 'string' && to.startsWith('#')) {
-      document.location.hash = to;
+      if (!to.substring(1)) {
+        // if only '#' symbol is found, remove the hash entirely.
```

Comment:
```suggestion
      if (to === '#') {
```

Would this be doing the same? It seems easier to understand?

Reply:
Yes indeed! That's much better.